readthedocs/templates/projects/import_config.html

```diff
@@ -0,0 +1,55 @@
+{% extends "projects/import_base.html" %}
+{% load i18n %}
+
+{% block content %}
+<h3>{% trans "Project configuration file (<code>.readthedocs.yaml</code>)" %}</h3>
+
+<p class="info">
+    {% blocktrans trimmed %}
+    Make sure your project has a <code>.readthedocs.yaml</code> at the root of your repository. This file is required by Read the Docs to be able to build your documentation. You can <a href="https://docs.readthedocs.io/en/stable/config-file/v2.html">read more about this in our documentation</a>.
+    {% endblocktrans %}
+</p>
+
+<p class="info">
+    Here you have an example for a common Sphinx project:
+
+    <pre style="padding-left: 1.5rem;font-size: 0.85rem;border: 1px solid;">
```

Comment:
I know that inline CSS is not optimal, but these are quick temporal changes and will be migrated soon to the new templates. See https://github.com/readthedocs/ext-theme/issues/165

Reply:
s/optimal/allowed/ :wink:

A better option is probably to reuse some of the CSS rules we already are using though. For example, you can leverage Pygments highlighting similar to how the webhook integration exchange output is displayed:

![image](https://github.com/readthedocs/readthedocs.org/assets/1140183/859fa26d-1a75-46cf-9e80-077fafe6053b)

I don't know if you can use `div.highlight > pre` without the wrapping element (`div.httpexchange`), but I would look at what rules do exist and reuse `div.httpexchange` or add another class just for the config file output. Either way, this gets you nicely highlighted text on both sites.

Reply:
It seems it's possible to use `div.highlight pre`, but it requires transforming the code with a helper in the model: https://github.com/readthedocs/readthedocs.org/blob/9d9226bc65d3380c4f3f74fa3730f10e00b59a2c/readthedocs/integrations/models.py#L183-L194

I don't think we can use this approach without over-engineering it. I will skip the highlight for now.

Reply:
> s/optimal/allowed/

Regarding using inline CSS, I don't understand why you said it's not allowed, but I can put it inside a `<style>` block in that page if that's better. I removed most of them since I had an old version of the compiled CSS in my S3 bucket and there were some updates. So, I only want the font to be smaller for now.

Reply:
>  I don't understand why you said it's not allowed

CSS styles applied via a style tag cannot be overridden with CSS stylesheets, and CSS inside HTML is awful to track down and maintain. So all CSS should be in CSS files instead of HTML templates -- and same for JS.

> It seems it's possible to use div.highlight pre, but it requires transforming the code with a helper

I feel like this is doing something fairly custom and isn't needed for your use case. All that helper is doing is reformatting unformatted JSON into indented JSON.

Untested, but you should be able to use something like this inside the template:

`<div class="highlight"><pre lang="yaml">{{ config_template }}</pre></div>`

If highlighting doesn't work, at least it has some of the default styles used in a code block -- padding, font, font size.

Reply:
> Nope, it uses a HtmlFormatter to generate a HTML with extra tags

Ahhhh yes, sorry my bad. I was thinking of highlight.js for some reason, which has front end code that renders these classes. Obviously Pygments doesn't do that.

We can come back to this to improve the styles later, they look fine for now. I do want a general pygments template filter for the new dashboard though. It seems easy to accomplish and use for cases like this.

Or we use highlight.js to keep this type of front end code out of Python.